bazel: use getops for stamp.sh

rail opened the pull request:
Previously, the stamp.sh script used positional arguments to handle its parameters. This commit changes it to use getopts, which is a more robust and flexible way to handle command-line options in shell scripts. This change improves the readability and maintainability of the script.

Additionally, the script now includes a new parameter, -d, which allows to disable telemetry.

Part of: RE-926
Release note: None

rail committed:
bazel: use getops for stamp.sh

Previously, the stamp.sh script used positional arguments to handle its
parameters. This commit changes it to use getopts, which is a more
robust and flexible way to handle command-line options in shell scripts.
This change improves the readability and maintainability of the script.

Additionally, the script now includes a new parameter, -d, which allows
to disable telemetry and publish the builds under a different name.

Part of: RE-926
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146755)
<!-- Reviewable:end -->

rickystewart reviewed (commented):
The actual stamp.sh changes look fine but there seems to be a bunch of telemetry-related release stuff going on in here that doesn't seem related to the point of the PR. I'm wondering if it's some sort of error/oversight?

dhartunian reviewed (commented):
Telemetry opt-out LGTM

rail committed:
crash_reporting: add a compile-time flag to disable crash reporting

Previously, there was a way to disable crash reporting, by disabling
telemetry. One of the downside of this approach is that startup crashes
are still reported, even if telemetry is disabled.

This change explicitly sets the crash report URL to an empty string in
case if telemetry disabled compile time.

Release note: none
Epic: none

rickystewart commented on pkg/util/log/logcrash/crash_reporting.go:
```diff
@@ -209,6 +210,22 @@ func PanicAsError(depth int, r interface{}) error {
 	return errors.NewWithDepthf(depth+1, "panic: %v", r)
 }
 
+// crashReportingDisabledString controls wether to opt out of crash reporting
```
whether, not wether

rail commented:
bors r=rickystewart
